File: java/0121_maxProfit.java
class Solution {
    public int maxProfit(int[] prices) {
        int hist=100000;
        int res=0;
        for(var i=0;i<prices.length;i++)
        {
            if(prices[i]<hist)
                hist = prices[i];
            res = Math.max(res,prices[i]-hist);
        }
        return res;
    }
}

File: cpp/0121_maxProfit.cpp
// 122是多次买卖，121是一次买卖
// 用一个hist记录当前的历史最低价格点
// 遍历计算最大利润
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int hist=100000;
        int res=0;
        for(auto p:prices)
        {
            if(p<hist)
                hist=p;
            res = max(res,p-hist);
        }
        return res;
    }
};
